ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ChatText.qml

```diff
@@ -78,24 +80,48 @@ Item {
         }
     }
 
+
+    Loader {
+        id: mask
+        anchors.fill: chatText
+        active: root.veryLongChatText
+        visible: false
+        sourceComponent: LinearGradient {
+            start: Qt.point(0, 0)
+            end: Qt.point(0, chatText.height)
+            gradient: Gradient {
+                GradientStop { position: 0.0; color: "white" }
+                GradientStop { position: 0.85; color: "white" }
+                GradientStop { position: 1; color: "transparent" }
+            }
+        }
+    }
+
+    Loader {
+        id: opMask
+        active: root.veryLongChatText && !root.readMore
+        anchors.fill: chatText
+        sourceComponent: OpacityMask {
+            source: chatText
+            maskSource: mask
+        }
+    }
+
     Loader {
         id: showMoreLoader
         active: root.veryLongChatText
         anchors.top: chatText.bottom
-        anchors.topMargin: Style.current.smallPadding
-        anchors.left: chatText.horizontalAlignment === Text.AlignLeft ? chatText.left : undefined
-        anchors.right: chatText.horizontalAlignment === Text.AlignLeft ? undefined : chatText.right
-
+        anchors.topMargin: - Style.current.padding
+        anchors.horizontalCenter: parent.horizontalCenter
         sourceComponent: Component {
-            StyledText {
-                text: root.readMore ?
-                          qsTr("Read less") :
-                          qsTr("Read more")
-                color: chatText.color
-                font.pixelSize: 12
-                font.underline: true
+            SVGImage {
+                id: emojiImage
+                width: 256
+                height: 44
+                fillMode: Image.PreserveAspectFit
+                source: "../../../../img/read-more.svg"
                 z: 100
-
+                transform: Rotation { origin.x: 128; origin.y: 22; angle: (root.readMore ? 0 : 180)}
```

Comment:
There is also a `rotation` property in Image. did you need `transform` to change the center?

Reply:
ah. I did not see that property in the page for Image. Must be inherited from Item